src/elements/stepper/stepper/stepper.ts

```diff
@@ -29,6 +30,24 @@ export
 class SbbStepperElement extends SbbHydrationMixin(LitElement) {
   public static override styles: CSSResultGroup = style;
 
+  /**
+   * If the sbb-stepper is used in a sbb-dialog, the marker on the selected element will not appear,
+   * because the calculations are done when the dialog is closed, so the marker has a width of 0;
+   * the same happens for the stepper height.
+   * We need to recalculate it when the element becomes visible.
+   */
+  private _observer = new IntersectionController(this, {
+    target: null,
+    callback: (entries) => {
+      entries.forEach((e) => {
+        if (e.intersectionRatio === 1) {
```

Comment:
would that also work, if the element is half visible? or should we use something like "> 0"?

Reply:
In this specific case, i see the interceptor going directly to 1 when the dialog is open; i feel we can relax the check with > 0